File: pkg/types/input.go
package types

type Text struct {
	Title       string `json:"title"`
	Type        string `json:"type"`
	Placeholder string `json:"placeholder,omitempty"`
	Default     string `json:"default,omitempty"`
}

type TextArea struct {
	Title       string `json:"title"`
	Type        string `json:"type"`
	Placeholder string `json:"placeholder,omitempty"`
	Default     string `json:"default,omitempty"`
}

type Checkbox struct {
	Title       string `json:"title"`
	Type        string `json:"type"`
	Label       string `json:"label,omitempty"`
	Placeholder string `json:"placeholder,omitempty"`
	Default     bool   `json:"default,omitempty"`
}

type Select struct {
	Title       string         `json:"title"`
	Type        string         `json:"type"`
	Placeholder string         `json:"placeholder,omitempty"`
	Default     any            `json:"default,omitempty"`
	Items       []DropDownItem `json:"items,omitempty"`
}

type DropDownItem struct {
	Title string `json:"title"`
	Value any    `json:"value"`
}

File: pkg/typescript/src/input.ts
type InputProps = {
    name: string;
    required: boolean;
}

type Textfield = InputProps & {
    type: "text";
    title: string;
    defaut?: string;
    placeholder?: string;
}

type TextArea = InputProps & {
    type: "textarea";
    title: string;
    defaut?: string;
    placeholder?: string;
}

type Password = InputProps & {
    type: "password";
    title: string;
    defaut?: string;
    placeholder?: string;
}

type Checkbox = InputProps & {
    type: "checkbox";
    label: string;
    title?: string;
    defaut?: boolean;
}

export type Input = Textfield | Password | Checkbox;
